addons/stock/models/stock_move.py

```diff
@@ -203,7 +204,23 @@ def _compute_product_uom(self):
         for move in self:
             move.product_uom = move.product_id.uom_id.id
 
-    @api.depends('picking_id', 'picking_id.location_dest_id')
+    @api.depends('picking_id.location_id')
+    def _compute_location_id(self):
+        for move in self:
+            if move.picked:
+                continue
```

Comment:
I would rather unpick the move in case of location change

Reply:
done